.github/workflows/deploy-to-github-pages.yml

```diff
@@ -0,0 +1,47 @@
+name: Build and Deploy Static Files to GitHub Pages
+on:
+  push:
+    branches: ["main"]
+
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+concurrency:
+  group: "pages"
+  cancel-in-progress: false
+
+jobs:
+  build:
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    runs-on: ubuntu-latest
+    steps:
```

Comment:
Why would we write a script in the repo and then just do it all again in this github action? The point of the scripts in the repo is to run the exact same scripts here. Same script, different context.

This job should basically just be "check out the code and run our script".

Reply:
Because the time difference in minutes _**per build**_, it is much better to use the actions provided by GitHub to install Node.